plugins/channel-safely/channel-manager.cpp

```diff
@@ -5,7 +5,17 @@
 #include <modules/EventManager.h> //hash function for df::coord
 #include <df/block_square_event_designation_priorityst.h>
 
+#define NUMARGS(...) std::tuple_size<decltype(std::make_tuple(__VA_ARGS__))>::value
+#define d_assert(condition, ...) \
+            static_assert(NUMARGS(__VA_ARGS__) >= 1, "d_assert(condition, format, ...) requires at least up to format as arguments"); \
+            if (!condition) {                                                   \
+                DFHack::Core::getInstance().getConsole().print(__VA_ARGS__);    \
```

Comment:
this should probably be `ERROR(manager).print(__VA_ARGS__)'`

Reply:
I was going to go with printerr like lethosor suggested on discord. Should I prefer this?

Reply:
looking at the code, no, not yet. use printerr so it goes to stderr. we need to modify the debug logging framework to also print to stderr for WARN and above

Reply:
Earlier today I had a brief thought about a plugin that would share stdout and stderr with a file log. I vaguely recall having written some code to do that, pretty sure it was some C code

Reply:
will you change this to printerr for this PR? it seems like output splitting would be a larger project

Reply:
Yea, my local has the commit. I was just waiting to boot back to linux to push